packages/design-system/src/components/web-components/ds-review/ds-review.tsx

```diff
@@ -0,0 +1,31 @@
+import { define } from '../preactement/define';
+import { Review } from '../../Review';
+
+const attributes = [
+  'class-name',
+  'edit-aria-label',
+  'edit-href',
+  'edit-text',
+  'heading',
+  'heading-level',
+] as const;
+
+const Wrapper = ({ children, ...otherProps }) => {
+  return <Review {...otherProps}>{children}</Review>;
+};
+
+/* eslint-disable @typescript-eslint/no-namespace */
+declare global {
+  namespace JSX {
+    interface IntrinsicElements {
+      'ds-review': JSX.IntrinsicElements['div'] & {
+        [K in (typeof attributes)[number]]?: string;
```

Comment:
This is a nit and an understanding check for myself:

Do you need this name spacing here? My understanding of this (which is likely incomplete/wrong) is that we're saying that the `ds-review` element is a div and that any attributes that are number types should be string. Is that correct? And if it is correct, I don't know that you need it since none of the Review component props take numbers?

Reply:
The number is actually referring to a numerical index in the `attributes` array as it uses it as an array of keys and basically iterates over it.

You know, at this point now that there are almost all of the web components that'll ever exist, we know the typing needs for all the component attributes and could probably create a generic type utility. It might not be worth the abstraction, but I was thinking something like this:

```ts
type WebComponent<Tag extends string, Attributes extends readonly string[], T = HTMLElement> = {
  [K in Tag]: React.DetailedHTMLProps<
    React.HTMLAttributes<T> & {
      [K in (Attributes)[number]]?: string;
    },
    T
  >;
};

/* eslint-disable @typescript-eslint/no-namespace */
declare global {
  namespace JSX {
    interface IntrinsicElements extends WebComponent<'ds-filter-chip', typeof attributes> {}
  }
}
/* eslint-enable */
```

Theoretically you could pass `HTMLDivElement` to the third optional parameter of the generic type for components with a div root.

Reply:
It really doesn't save much boilerplate, though, or make it easier to read.

Reply:
Thanks for the clarification @pwolfert !